### ISSUE TITLE
attributes: make parameter fields optional

### ISSUE THREAD

**Issue by couchand** [tokio-rs/tracing#396]
### Crates

* `tracing-attributes`

### Motivation

The `#[instrument]` annotation is just about the most pleasant way I've ever seen to trace code (particularly with the async-await feature flag enabled!).  While usually it's quite nice to automatically include function arguments as fields, there are several ergonomic concerns:

* only "normal" `Ident` patterns are included,
* argument types must implement `fmt::Debug`, and
* some `Debug` impls produce extremely long values.

It would be nice to have some measure of control over this.  Trying to solve the first may very well open a whole can of worms (perhaps it deserves its own issue...), but the second and third can be addressed by providing some way to opt out parameters.

### Proposal

Provide some way to prevent certain parameters from being included as fields on the span.  Two strategies come immediately to mind:

* A white list.  By explicitly naming the parameters that *should* be fields, we can exclude those that shouldn't.
* A black list.  We state directly the parameters that shouldn't be included as fields.

The primary advantage I see to the former is that the macro arg name can be more obvious.  For instance:

```rust
#[instrument(params=[])]
fn do_it(parameter: TypeWithoutDebug) {}
```

### Alternatives

Keep on writing the macro invocation explicitly everywhere the default behavior is insufficient.  A bit frustrating, but not too difficult.

**Comment by couchand** [comment]
I see now that this was already covered in #11 and closed in #359, which I missed due to the fact that `tracing-attributes` version 0.1.4 hasn't yet been included in a top-level `tracing` release.  Forgive the unnecessary issue.

**Comment by hawkw** [comment]
We should probably get a release of `tracing` out which includes #359.
